solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -897,6 +902,15 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
         uint256 poolAmount,
         uint256 reserveAmount
     ) internal returns (uint256) {
+        (
+            uint256 spotRateN,
+            uint256 spotRateD,
+            uint256 averageRateN,
+            uint256 averageRateD
+        ) = reserveTokenRates(poolToken, reserveToken);
```

Comment:
Perhaps `reserveTokenRates` can return a struct which `verifyRate` can then verify, which can be also refactored to `PackedRates`? It can be also a good opportunity to switch to `Fraction` (and `Fraction128` for `PackedRates`) so that we won't keep carrying around all these variables.

Reply:
We can also have an overload for `verifyRate` that received `PackedRates` as well

Reply:
done